### ISSUE TITLE
Allow marker optimization=false in module configuration

### ISSUE THREAD

**Issue by fabn** [allenhwkim/angularjs-google-maps#295]
The only reliable way to write an integration test for google maps v3 seems to be [disabling marker optimization](http://blog.mojotech.com/selecting-google-maps-v3-markers-with-selenium-or-jquery/) in order to interact with markers using jQuery (or capybara/webdriver) selectors. I know your library already permit this by using attributes i.e. 

```html
<marker postion='NY' optimized=false>
```

I was wondering if you can provide a configurable service (Attr2Options??) to globally disable optimization, in this way instead of cluttering app code with `optimized="{{whateverFlag}}"` one can simply define it in module initialization, something like:

```js
angular.module('myApp', ["ngMap"]).config(function(NgOptimizerProvider) {
  NgOptmizerProvider.disableMarkerOptimization(); // or whatever names you choose
});
```

It should be pretty easy to do and imho would be very useful. What do you think?

**Comment by allenhwkim** [comment]
Good idea, but what could it be? only for a marker? or including all other directives?
The implementation would be simple, but design should be done properly at early stage.

Need few options that requires configuration.
  1. Default marker options

**Comment by fabn** [comment]
According to [documentation](https://developers.google.com/maps/documentation/javascript/reference) optimized option is supported only for markers.

In order to create an extensible solution I'd start with a NgMapOptions configurable provider open to future improvements. This provider will be then used by Attr2Options service to fetch global options or by individual directives.

Right now I don't know what other global options could go in that provider.

**Comment by allenhwkim** [comment]
`NgMap` factory is introduced in 1.14, we can expand this `NgMap` as a provider later.

**Comment by allenhwkim** [comment]
This has been implemented with 1.14.6. Example here, http://ngmap.github.io/#/marker-with-default-options.html
